api.go

```diff
@@ -140,10 +125,30 @@ func (conn *Connection) rqliteApiPost(apiOp apiOperation, sqlStatements []string
 
 	trace("%s: rqliteApiPost() called for a QUERY of %d statements", conn.ID, len(sqlStatements))
 
-	// Send statements as json
-	body, err := json.Marshal(sqlStatements)
+	if len(parameters) == 0 {
+		// Send statements as json
+		body, err := json.Marshal(sqlStatements)
+		if err != nil {
+			return nil, err
+		}
+		return conn.rqliteApiCall(ctx, apiOp, "POST", body)
+	}
+
+	// If the parameters exists, whether is a single query or a batch,
+	// we should use the bulk API.
+	var bulkParameterizedBuilder [][]interface{}
+	for _, stmt := range sqlStatements {
+		bulkParameterizedBuilder = append(bulkParameterizedBuilder, []interface{}{stmt})
+	}
+
+	for i, parameter := range parameters {
```

Comment:
This loop could cause an "index out of range" error/panic. IMO, there should be some validation that the length of the `parameters` array matches the length of the `sqlStatements` array when parameters are passed.